### PR TITLE
feat(#78): implement useShippingProvider composable

### PR DESCRIPTION
**Pull request by danhaj**
## Related Issue

#78 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rafalcymerys** [comment]
@danhaj in CartPreview.vue there's a hardcoded getShippingMethodPrice method.

We should remove it and replace it's use with `cartGetters.getShippingPrice(cart);`

File: packages/composables/src/getters/checkoutGetters.ts [hunk 1]
```diff
@@ -1,16 +1,14 @@
 import { ShippingMethod } from '@vue-storefront/spree-api';
```
Comment (rafalcymerys) [inline packages/composables/src/getters/checkoutGetters.ts:1]: Is this import used anywhere now?

File: packages/composables/src/useShippingProvider/index.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
+    const shipments = await context.$spree.api.getShipments();
+    return shipments;
   },
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  save: async (context: Context, { shippingMethod, customQuery }) => {
-    console.log('Mocked: saveShippingProvider');
-    provider = shippingMethod;
-    return provider;
+  save: async (context: Context, { shippingMethod }) => {
```
Comment (rafalcymerys) [inline packages/composables/src/useShippingProvider/index.ts:14]: Is it possible to change the params signature of this method? `shippingMethod` is not really what we pass here.
Is that defined by VSF?

File: packages/theme/components/Checkout/ShippingRatePicker.vue [hunk 3]
```diff
@@ -8,11 +8,11 @@
     <SfRadio
       v-e2e="'shipping-method'"
       v-for="rate in shippingRates"
-      :key="rate.id"
-      :label="rate.name"
-      :value="rate.id"
-      :description="$n(rate.cost, 'currency')"
-      :selected ="selectedRateId"
+      :key="checkoutGetters.getShippingMethodId(rate)"
```
Comment (rafalcymerys) [inline packages/theme/components/Checkout/ShippingRatePicker.vue:11]: Good stuff :)

File: packages/composables/src/getters/checkoutGetters.ts [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
+import { ShippingRate } from '@vue-storefront/spree-api/src/types';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const getShippingMethodId = (shippingMethod: ShippingMethod): string => '';
+export const getShippingMethodId = (shippingMethod: ShippingRate): string => shippingMethod.id;
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const getShippingMethodName = (shippingMethod: ShippingMethod): string => '';
+export const getShippingMethodName = (shippingMethod: ShippingRate): string => shippingMethod.name;
```
Comment (rafalcymerys) [inline packages/composables/src/getters/checkoutGetters.ts:6]: Looking at e.g. productGetters - shouldn't we also handle shippingMethod=undefined here? And in such case return empty string or something?
The same for other getters :)

Comment (danhaj) [reply packages/composables/src/getters/checkoutGetters.ts:6]: I think `shippingMethod` shouldn't be undefined (if there are no shipping methods - we are not calling these getters), although we can add that to be 100% sure

File: packages/composables/src/getters/checkoutGetters.ts (hunk 1 above)
Comment (danhaj) [reply packages/composables/src/getters/checkoutGetters.ts:1]: I don't think so, we can remove it

**Review by rafalcymerys — APPROVED** [review]

**Pull request merged**